### ISSUE TITLE
Player Stuck when talking to NPC

### ISSUE THREAD

**Issue by karlseven** [rathena/rathena#2348]
* **rAthena Hash**: 
2d31bc3

* **Client Date**: 

2015-11-04aRagexe

* **Server Mode**: 

Renewal

* **Description of Issue**: 
  * Result: Player stuck on some cases when talking to stylist or warper
  * Expected Result: Should not be stuck when browsing color/style list or choose warp location
  * How to Reproduce: Just talk to stylist or warper like 10 times
  * Official Information: None. These are custom NPC

* **Modifications that may affect results**: 
None

**Comment by anacondaqq** [comment]
@karlseven can you please add a script with comments where it stuck to make devs know is it a script engine problem, or just wrong written script?

**Comment by Jeybla** [comment]
@karlseven is this issue related #2177 ?

**Comment by karlseven** [comment]
@anacondaqq 
the warper and stylist that is included with rathena, it seems npc scripts with lists, im not sure
@Jeybla Yes somehow that happens also
they say its about long list. i dont really know but only those 2 scripts like warper and stylist that comes from rathena has that problem

**Comment by hurtsky** [comment]
@karlseven are you using another custom script like hourly reward?

**Comment by Jeybla** [comment]
- Do you encounter this issue only after you logged in?
From anacondaqq: `I can't reproduce the issue if I log in and doing something in the game for a while.`

Related Scripts:
https://github.com/rathena/rathena/blob/master/npc/custom/stylist.txt
https://github.com/rathena/rathena/blob/master/npc/custom/warper.txt
Both NPCs don't have a mes-Window. An workaround would be to add a `mes`. The menu would still vanish, but you'd be able to press the `close`-Button so you don't get stuck.

Anyway this very likely is a duplicate of 2177. I'll close this issue; but it would be great to have some more detailed info about it. Thanks

**Comment by karlseven** [comment]
@hurtsky 
Yes im using a script that gives you reward hourly. and a floating rates script.

**Comment by hurtsky** [comment]
@karlseven it caused by that hourly reward script, try disable/remove that script.. and you wont get that bug anymore..

**Comment by Lemongrass3110** [comment]
@karlseven could you confirm that this is caused by this script?
@hurtsky do you happen to know what part of this script is causing this kind of error to happen? Could you send it to us(discord or forum) so we could analyze it?

**Comment by hurtsky** [comment]
@Lemongrass3110 im pretty sure he using this hourly script
https://rathena.org/board/topic/109804-hourly-reward-item/

im not sure what causing it, but probably, i think because it using
 `attachnpctimer ""+strcharinfo(0)+"";`

if the timer run while the player talking to npc, the current npc dialog windows will close and that player will stuck (the solution just logout and relog) but it will keep happen like that.

**Comment by karlseven** [comment]
@hurtsky 
yes i have a feeling that's the part that make the script hang or the player stops. i tried removing each npc that has that function and the problem never happens

**Comment by keitenai** [comment]
@Lemongrass3110  I'm not using hourly event but Even tool dealer makes players stucked

![capture](https://user-images.githubusercontent.com/11855571/30093087-9e29a968-92f5-11e7-9ca3-ca4dae65cfbb.PNG)

**Comment by Jeybla** [comment]
Duplicate of #2177. Closing this issue.
